osu.Game/Overlays/ChatOverlay.cs

```diff
@@ -288,6 +318,7 @@ private void currentChannelChanged(ValueChangedEvent<Channel> e)
             // mark channel as read when channel switched
             if (e.NewValue.Messages.Any())
                 channelManager.MarkChannelAsRead(e.NewValue);
+            CheckIsLoggedIn();
```

Comment:
Do these checks really need to be all over the place here? Isn't the API state change enough?

Reply:
In fact, if the checkIsLoggedIn() method is not called, the chat will be displayed when the user will change channel even if the user is not logged in, because there is no status change.